integration/helpers.go

```diff
@@ -1188,7 +1188,7 @@ func (s *discardServer) Stop() {
 	s.sshServer.Close()
 }
 
-func (s *discardServer) HandleNewChan(conn net.Conn, sconn *ssh.ServerConn, newChannel ssh.NewChannel) {
+func (s *discardServer) HandleNewChan(conn *utils.TrackingConn, sconn *ssh.ServerConn, newChannel ssh.NewChannel) {
```

Comment:
I suggest to keep `net.Conn` as a primary interface here and below, to avoid pinning our library to such a narrow struct spec.